### PR TITLE
Sandbox: add a development mode

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
With this PR, sandbox reject non development version of LF and transaction.
Previous behaviour can be set using the  command line option `--dev-mode`.

This is part of #5164.

CHANGELOG_BEGIN
 * [Sandbox] By default Sandbox rejects the development versions of
   DAML-LF and transaction format. One has to explicitly use the
   command line option `--dev-mode` to allows those versions.
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala [hunk 1]
```diff
@@ -97,4 +40,23 @@ object EngineConfig {
...
 
+  // development configuration, should not be used in PROD.
+  // accept all language and transaction versions supported by SDK_1_x plus development versions.
+  val Dev: EngineConfig = toDev(Stable)
+
+  // Legacy configuration, to be used by sandbox classic only
+  @deprecated("Sandbox_Classic_Stable is to be used by sandbox classic only", since = "1.5.0")
+  val Sandbox_Classic_Stable: EngineConfig =
+    Stable.copy(
+      languageVersions = Stable.languageVersions.copy(min = LV(LV.Major.V0, LV.Minor.Stable(""))),
```
Comment (SamirTalwar-DA) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala:51]: This seems to have changed from v1.1 to v0. Was this intentional?

Comment (remyhaemmerle-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/EngineConfig.scala]: Yes. But, it does not really matter as we does not really support V0.
I should probably remove it.

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala [hunk 2]
```diff
@@ -63,14 +63,19 @@ object SandboxServer {
...
+
+  private def getEngine(config: EngineConfig): Engine = synchronized {
+    engine match {
+      case Some(eng) if eng.config == config => eng
+      case _ =>
+        val eng = new Engine(config)
+        engine = Some(eng)
+        eng
+    }
+  }
```
Comment (remyhaemmerle-da) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala:78]: @SamirTalwar-DA what do you think about that ?

Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala:78]: Looks good to me. 😀

**Pull request merged**